src/app/src/test/kotlin/org/imperial/mrc/hint/unit/controllers/ADRControllerTests.kt

```diff
@@ -328,32 +370,147 @@ class ADRControllerTests : HintrControllerTests()
+                            "resource_type" to resourceType,
+                            "id" to "testResId")),
+                    any()) } doReturn ResponseEntity.ok().body("whatever")
+        }
+        val mockBuilder: ADRClientBuilder = mock {
+            on { build() } doReturn mockClient
+        }
+
+        val sut = ADRController(mock(), mock(), mockBuilder, mock(), mockProperties, mockFileManager, mock(), mock(), mock())
+        val result = sut.pushFileToADR("datasetId", resourceType, "calId",
+                "testResFilename.", "testResName", "testResId")
+
+        assertThat(result.statusCode).isEqualTo(HttpStatus.OK)
+        assertThat(result.body!!).isEqualTo("whatever")
+
+        val argumentCaptor = argumentCaptor<Pair<String, File>>()
+        verify(mockClient).postFile(any(), any(), argumentCaptor.capture())
+        assertThat(argumentCaptor.firstValue.first).isEqualTo("upload")
+        val uploadedFile = argumentCaptor.firstValue.second
+        assertThat(uploadedFile.name).isEqualTo("original-test.$fileExt")
+    }
+
+    @Test
+    fun `pushes pjnz to ADR`()
+    {
+        testPushesInputFile(FileType.PJNZ, "pjnz", "adr-pjnz")
+    }
+
+    @Test
+    fun `pushes shape file to ADR`()
+    {
+        testPushesInputFile(FileType.Shape, "geojson", "adr-shape")
+    }
+
+    @Test
+    fun `pushes population file to ADR`()
+    {
+        testPushesInputFile(FileType.Population, "csv", "adr-pop")
+    }
+
+    @Test
+    fun `pushes survey file to ADR`()
+    {
+        testPushesInputFile(FileType.Survey, "csv", "adr-survey")
+    }
+
+    @Test
+    fun `pushes art file to ADR`()
+    {
+        testPushesInputFile(FileType.Programme, "csv", "adr-art")
+    }
+
+    @Test
+    fun `pushes anc file to ADR`()
+    {
+        testPushesInputFile(FileType.ANC, "csv", "adr-anc")
+    }
+
+    @Test
+    fun `returns error on upload input file without resourceId`()
+    {
+        val sut = ADRController(mock(), mock(), mock(), mock(), mockProperties, mock(), mock(), mock(), mock())
+        val result = sut.pushFileToADR("datasetId", "adr-pjnz", "calId",
+                "testResFilename.", "testResName", null)
+        assertThat(result.statusCode).isEqualTo(HttpStatus.BAD_REQUEST)
+        assertThat(result.body!!).contains("resourceId must be provided for input resourceType")
+    }
+
+    @Test
+    fun `returns error on upload input file which does not exist in version`()
+    {
+        val sut = ADRController(mock(), mock(), mock(), mock(), mockProperties, mock(), mock(), mock(), mock())
+        val result = sut.pushFileToADR("datasetId", "adr-pjnz", "calId",
+                "testResFilename.", "testResName", "testResId")
+        assertThat(result.statusCode).isEqualTo(HttpStatus.BAD_REQUEST)
+        assertThat(result.body!!).contains("File does not exist")
+    }
+
+    @Test
+    fun `throws exception on make input file to push to ADR if not input resource type`()
```

Comment:
Adding this to satisfy code coverage by covering the default line of the `when` mapping string resource type to file type in the private `makeInputFileToPushToADR` method (which is there to break up the long `pushFileToADR` method to keep linting happy!). Logically, it shouldn't ever (!) hit that line, given the way the `pushFileToADR` distinguishes input from output files, hence using reflection here to call the private method separately. 

There may be a more elegant way to deal with all this, and I think changes in Lekan's metadata branch is also going to result in some restructuring of the method so should possibly await the outcome of that before settling on this approach.